dl4j-examples/src/main/java/org/deeplearning4j/examples/nlp/sentencepiecernnexample/SentencePieceRNNExample.java

```diff
@@ -0,0 +1,260 @@
+import org.nd4j.linalg.dataset.api.MultiDataSet;
+import org.nd4j.linalg.dataset.api.MultiDataSetPreProcessor;
+import org.nd4j.linalg.learning.config.Adam;
+import org.nd4j.linalg.lossfunctions.LossFunctions;
+import org.deeplearning4j.examples.utilities.DataUtilities;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+/**
+ * @author andrewtuzhykov@gmail.com
+ */
+
+    /**
+     * NOTE: You may download already trained defined below model for your own inference
+     *  https://dl4jdata.blob.core.windows.net/dl4j-examples/models/sentencepiece_rnn_example_model.zip
+     */
+
+
+public class SentencePieceRNNExample {
+
+
+    /**
+     * Data URL for downloading
+     */
+    public static final String DATA_URL = "http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz";
+
+    // Bert Base Uncased Vocabulary
+    public static final String VOCAB_URL = "https://s3.amazonaws.com/models.huggingface.co/bert/bert-large-uncased-vocab.txt";
+
+    /**
+     * Location to save and extract the training/testing data
+     */
+    public static final String DATA_PATH = FilenameUtils.concat(System.getProperty("java.io.tmpdir"), "dl4j_w2vSentiment/");
+
+
+    /**
+     * Get BertIterator instance.
+     *
+     * @param isTraining specifies which dataset iterator we want to get: train or test.
+     * @param t          BertWordPieceTokenizerFactory initialized with provided vocab.
+     * @return BertIterator with specified parameters.
+     */
+
+    public static BertIterator getBertDataSetIterator(boolean isTraining, BertWordPieceTokenizerFactory t) {
+
+        String path = FilenameUtils.concat(DATA_PATH, (isTraining ? "aclImdb/train/" : "aclImdb/test/"));
+        String positiveBaseDir = FilenameUtils.concat(path, "pos");
+        String negativeBaseDir = FilenameUtils.concat(path, "neg");
+        Random rng = new Random(42);
+
+        File filePositive = new File(positiveBaseDir);
+        File fileNegative = new File(negativeBaseDir);
+
+        Map<String, List<File>> reviewFilesMap = new HashMap<>();
+        reviewFilesMap.put("Positive", Arrays.asList(Objects.requireNonNull(filePositive.listFiles())));
+        reviewFilesMap.put("Negative", Arrays.asList(Objects.requireNonNull(fileNegative.listFiles())));
+
+
+        BertIterator b = BertIterator.builder()
+            .tokenizer(t)
+            .lengthHandling(BertIterator.LengthHandling.FIXED_LENGTH, 256)
+            .minibatchSize(32)
+            .sentenceProvider(new FileLabeledSentenceProvider(reviewFilesMap, rng))
+            .featureArrays(BertIterator.FeatureArrays.INDICES_MASK)
+            .vocabMap(t.getVocab())
+            .task(BertIterator.Task.SEQ_CLASSIFICATION)
+            .build();
+
+
+        return b;
+    }
+
+
+    public static void main(String[] args) throws Exception {
```

Comment:
Let's rearrange things a bit - put the main method above the getBertDataSetIterator method in the class, for better UX
Users probably want to see the main method first - whereas right now they have to scroll past the data method before they see the starting point and main code